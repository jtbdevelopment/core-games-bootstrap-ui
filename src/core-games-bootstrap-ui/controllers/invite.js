'use strict';

angular.module('coreGamesBootstrapUi.controllers').controller('CoreBootstrapInviteCtrl',
    ['$uibModalInstance', 'invitableFriends', 'message', 'jtbFacebook',
        function ($uibModalInstance, invitableFriends, message, jtbFacebook) {
            var controller = this;
            controller.invitableFriends = invitableFriends;
            controller.chosenFriends = [];
            controller.message = message;
            controller.invite = function () {
                var ids = [];
                angular.forEach(controller.chosenFriends, function (chosen) {
                    ids.push(chosen.id);
                });
                jtbFacebook.inviteFriends(ids, message);
                $uibModalInstance.close();
            };
            controller.cancel = function () {
                $uibModalInstance.dismiss();
            };
        }]);

