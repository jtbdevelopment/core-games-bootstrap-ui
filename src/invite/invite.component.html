<div class="invite-dialog">
    <div class="modal-header">
        <h3 class="modal-title">Invite Friends to Play</h3>
    </div>
    <div class="modal-body">
        <p-multiSelect [options]="invitable"
                       dataKey="id"
                       optionLabel="displayName"
                       maxSelectedLabels="30"
                       [(ngModel)]="chosen"
                       [ngModelOptions]="{standalone: true}">
        </p-multiSelect>
    </div>
    <div class="modal-footer">
        <button class="btn btn-primary invite-friends" (click)="inviteFriends()">Invite</button>
        <button class="btn btn-warning cancel-invite" (click)="cancel()">Cancel</button>
    </div>
</div>
