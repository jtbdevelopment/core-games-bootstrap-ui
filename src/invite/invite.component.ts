import {Component, Inject} from '@angular/core';
import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
import {AppConfig, FacebookInviteService, FriendsService, Invitable} from 'jtb-core-games-ui';

@Component({
    selector: 'invite-friends',
    template: require('./invite.component.html'),
})
export class InviteComponent {
    public invitable: Invitable[] = [];
    public chosen: Invitable[] = [];

    constructor(private facebookInviteService: FacebookInviteService,
                private friendService: FriendsService,
                private modalRef: NgbActiveModal,
                @Inject('AppConfig') private config: AppConfig) {
        this.friendService.invitableFriends.subscribe(f => {
            this.invitable = f;
            this.chosen = [];
        });
    }

    public cancel(): void {
        this.modalRef.dismiss();
    }

    public inviteFriends(): void {
        this.facebookInviteService.inviteFriends(this.chosen, this.config.inviteFriendsMessage);
        this.modalRef.close();
    }
}
