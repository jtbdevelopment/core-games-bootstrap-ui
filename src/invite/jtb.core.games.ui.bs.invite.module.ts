import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
import {InviteComponent} from './invite.component';
import {MultiSelectModule} from 'primeng/primeng';
import {FormsModule} from '@angular/forms';
import {JTBCoreGamesUI} from 'jtb-core-games-ui';

@NgModule({
    imports: [
        BrowserModule,
        NgbModule,
        FormsModule,
        MultiSelectModule,
        JTBCoreGamesUI
    ],
    providers: [
        {provide: 'Window', useValue: window}
    ],
    entryComponents: [
        InviteComponent
    ],
    declarations: [
        InviteComponent
    ]
})
export class JTBCoreGamesUIBInviteModule {
}

