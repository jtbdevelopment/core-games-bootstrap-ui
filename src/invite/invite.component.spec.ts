import {ComponentFixture, TestBed} from '@angular/core/testing';
import {BehaviorSubject} from 'rxjs/BehaviorSubject';
import {Observable} from 'rxjs/Observable';
import {InviteComponent} from './invite.component';
import {NgbActiveModal, NgbModule} from '@ng-bootstrap/ng-bootstrap';
import {MultiSelectModule} from 'primeng/primeng';
import {FormsModule} from '@angular/forms';
import {AppConfig, FacebookInviteService, FriendsService, Invitable} from 'jtb-core-games-ui';

class MockFacebookInviteService {
    inviteFriends = jasmine.createSpy('inviteFriends');
}

class MockFriendService {
    public invitableFriendSubject: BehaviorSubject<Invitable[]> = new BehaviorSubject<Invitable[]>([]);
    public invitableFriends: Observable<Invitable[]> = Observable.from<Invitable[]>(this.invitableFriendSubject);
}

class MockConfig implements AppConfig {
    releaseNotes: string = '';
    hoverMenu: boolean = false;
    appName: string = '';
    version: string = '';
    inviteFriendsMessage = 'come play games!';
}

class MockModal {
    dismiss = jasmine.createSpy('dismiss');
    close = jasmine.createSpy('close');
}


describe('Component:  invite component', () => {
    let fbInvite: MockFacebookInviteService;
    let friendService: MockFriendService;
    let modalRef: MockModal;
    let fixture: ComponentFixture<InviteComponent>;

    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                NgbModule,
                FormsModule,
                MultiSelectModule
            ],
            providers: [
                {provide: FacebookInviteService, useClass: MockFacebookInviteService},
                {provide: FriendsService, useClass: MockFriendService},
                {provide: NgbActiveModal, useClass: MockModal},
                {provide: 'AppConfig', useClass: MockConfig}
            ],
            declarations: [
                InviteComponent
            ]
        });
        TestBed.compileComponents();
        modalRef = TestBed.get(NgbActiveModal);
        fixture = TestBed.createComponent(InviteComponent);
        fbInvite = TestBed.get(FacebookInviteService);
        friendService = TestBed.get(FriendsService);
    });

    //  Not testing prime faces multi-select

    it('subscribes to friends and updates clear chosen', () => {
        fixture.componentInstance.chosen = [new Invitable('d3', 'n3')];
        let friends = [new Invitable('d1', 'n1'), new Invitable('d2', 'n2')];
        friendService.invitableFriendSubject.next(friends);
        fixture.detectChanges();
        expect(fixture.componentInstance.chosen).toEqual([]);
        expect(fixture.componentInstance.invitable).toEqual(friends);
    });

    it('cancel closes and does not invite', () => {
        fixture.nativeElement.querySelector('.cancel-invite').click();
        expect(modalRef.dismiss).toHaveBeenCalled();
        expect(fbInvite.inviteFriends).not.toHaveBeenCalled();
    });

    it('invite invites the chosen', () => {
        let friends = [new Invitable('d1', 'n1'), new Invitable('d2', 'n2'), new Invitable('x1', 'x3')];
        friendService.invitableFriendSubject.next(friends);
        fixture.componentInstance.chosen = friends;
        fixture.detectChanges();
        fixture.nativeElement.querySelector('.invite-friends').click();
        expect(fbInvite.inviteFriends).toHaveBeenCalledWith(friends, 'come play games!');
        expect(modalRef.close).toHaveBeenCalled();
    });
});
