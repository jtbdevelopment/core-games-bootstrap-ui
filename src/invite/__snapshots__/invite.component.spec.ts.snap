// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  invite component invite invites the chosen 1`] = `
<invite-friends
  chosen={[Function Array]}
  config={[Function MockConfig]}
  facebookInviteService={[Function MockFacebookInviteService]}
  friendService={[Function MockFriendService]}
  invitable={[Function Array]}
  modalRef={[Function MockModal]}
>
  <div
    class="invite-dialog"
  >
    
      
    <div
      class="modal-header"
    >
      
          
      <h3
        class="modal-title"
      >
        Invite Friends to Play
      </h3>
      
      
    </div>
    
      
    <div
      class="modal-body"
    >
      
          
      <p-multiselect
        class="ng-untouched ng-pristine ng-valid"
        datakey="id"
        maxselectedlabels="30"
        ng-reflect-data-key="id"
        ng-reflect-max-selected-labels="30"
        ng-reflect-model="[object Object],[object Object"
        ng-reflect-option-label="displayName"
        ng-reflect-options="[object Object]"
        optionlabel="displayName"
      >
        
          
        <div
          class="ui-multiselect ui-widget ui-state-default ui-corner-all"
          ng-reflect-ng-class="[object Object]"
        >
          
              
          <div
            class="ui-helper-hidden-accessible"
          >
            
                  
            <input
              readonly="readonly"
              type="text"
            />
            
              
          </div>
          
              
          <div
            class="ui-multiselect-label-container"
            title="Choose"
          >
            
                  
            <label
              class="ui-multiselect-label ui-corner-all"
            >
              Choose
            </label>
            
              
          </div>
          
              
          <div
            class="ui-multiselect-trigger ui-state-default ui-corner-right"
            ng-reflect-ng-class="[object Object]"
          >
            
                  
            <span
              class="ui-clickable fa fa-fw fa-caret-down"
              ng-reflect-klass="ui-clickable"
              ng-reflect-ng-class="fa fa-fw fa-caret-down"
            />
            
              
          </div>
          
              
          <div
            class="ui-multiselect-panel ui-widget ui-widget-content ui-corner-all ui-shadow"
            ng-reflect-ng-class="ui-multiselect-panel ui-widget"
            style="display: none;"
          >
            
                  
            <div
              class="ui-widget-header ui-corner-all ui-multiselect-header ui-helper-clearfix"
              ng-reflect-klass="ui-widget-header ui-corner-all"
              ng-reflect-ng-class="[object Object]"
            >
              
                      
              
              <div
                class="ui-chkbox ui-widget"
              >
                
                          
                <div
                  class="ui-helper-hidden-accessible"
                >
                  
                              
                  <input
                    readonly="readonly"
                    type="checkbox"
                  />
                  
                          
                </div>
                
                          
                <div
                  class="ui-chkbox-box ui-widget ui-corner-all ui-state-default"
                  ng-reflect-klass="ui-chkbox-box ui-widget ui-cor"
                  ng-reflect-ng-class="[object Object]"
                >
                  
                              
                  <span
                    class="ui-chkbox-icon ui-clickable"
                    ng-reflect-klass="ui-chkbox-icon ui-clickable"
                    ng-reflect-ng-class="[object Object]"
                  />
                  
                          
                </div>
                
                      
              </div>
              
                      
              
              <div
                class="ui-multiselect-filter-container"
              >
                
                          
                <input
                  class="ui-inputtext ui-widget ui-state-default ui-corner-all"
                  role="textbox"
                  type="text"
                />
                
                          
                <span
                  class="fa fa-fw fa-search"
                />
                
                      
              </div>
              
                      
              <a
                class="ui-multiselect-close ui-corner-all"
                href="#"
              >
                
                          
                <span
                  class="fa fa-close"
                />
                
                      
              </a>
              
                  
            </div>
            
                  
            <div
              class="ui-multiselect-items-wrapper"
            >
              
                      
              <ul
                class="ui-multiselect-items ui-multiselect-list ui-widget-content ui-widget ui-corner-all ui-helper-reset"
                style="max-height: 200px;"
              >
                
                          
                
                <li
                  class="ui-multiselect-item ui-corner-all"
                  ng-reflect-klass="ui-multiselect-item ui-corner-"
                  ng-reflect-ng-class="[object Object]"
                  style="display: block;"
                >
                  
                              
                  <div
                    class="ui-chkbox ui-widget"
                  >
                    
                                  
                    <div
                      class="ui-helper-hidden-accessible"
                    >
                      
                                      
                      <input
                        aria-label="n1"
                        readonly="readonly"
                        type="checkbox"
                      />
                      
                                  
                    </div>
                    
                                  
                    <div
                      class="ui-chkbox-box ui-widget ui-corner-all ui-state-default"
                      ng-reflect-klass="ui-chkbox-box ui-widget ui-cor"
                      ng-reflect-ng-class="[object Object]"
                    >
                      
                                      
                      <span
                        class="ui-chkbox-icon ui-clickable"
                        ng-reflect-klass="ui-chkbox-icon ui-clickable"
                        ng-reflect-ng-class="[object Object]"
                      />
                      
                                  
                    </div>
                    
                              
                  </div>
                  
                              
                  
                  <label>
                    n1
                  </label>
                  
                              
                  
                  
                          
                </li>
                <li
                  class="ui-multiselect-item ui-corner-all"
                  ng-reflect-klass="ui-multiselect-item ui-corner-"
                  ng-reflect-ng-class="[object Object]"
                  style="display: block;"
                >
                  
                              
                  <div
                    class="ui-chkbox ui-widget"
                  >
                    
                                  
                    <div
                      class="ui-helper-hidden-accessible"
                    >
                      
                                      
                      <input
                        aria-label="n2"
                        readonly="readonly"
                        type="checkbox"
                      />
                      
                                  
                    </div>
                    
                                  
                    <div
                      class="ui-chkbox-box ui-widget ui-corner-all ui-state-default"
                      ng-reflect-klass="ui-chkbox-box ui-widget ui-cor"
                      ng-reflect-ng-class="[object Object]"
                    >
                      
                                      
                      <span
                        class="ui-chkbox-icon ui-clickable"
                        ng-reflect-klass="ui-chkbox-icon ui-clickable"
                        ng-reflect-ng-class="[object Object]"
                      />
                      
                                  
                    </div>
                    
                              
                  </div>
                  
                              
                  
                  <label>
                    n2
                  </label>
                  
                              
                  
                  
                          
                </li>
                <li
                  class="ui-multiselect-item ui-corner-all"
                  ng-reflect-klass="ui-multiselect-item ui-corner-"
                  ng-reflect-ng-class="[object Object]"
                  style="display: block;"
                >
                  
                              
                  <div
                    class="ui-chkbox ui-widget"
                  >
                    
                                  
                    <div
                      class="ui-helper-hidden-accessible"
                    >
                      
                                      
                      <input
                        aria-label="x3"
                        readonly="readonly"
                        type="checkbox"
                      />
                      
                                  
                    </div>
                    
                                  
                    <div
                      class="ui-chkbox-box ui-widget ui-corner-all ui-state-default"
                      ng-reflect-klass="ui-chkbox-box ui-widget ui-cor"
                      ng-reflect-ng-class="[object Object]"
                    >
                      
                                      
                      <span
                        class="ui-chkbox-icon ui-clickable"
                        ng-reflect-klass="ui-chkbox-icon ui-clickable"
                        ng-reflect-ng-class="[object Object]"
                      />
                      
                                  
                    </div>
                    
                              
                  </div>
                  
                              
                  
                  <label>
                    x3
                  </label>
                  
                              
                  
                  
                          
                </li>
                
                      
              </ul>
              
                  
            </div>
            
              
          </div>
          
          
        </div>
        
      
      </p-multiselect>
      
      
    </div>
    
      
    <div
      class="modal-footer"
    >
      
          
      <button
        class="btn btn-primary invite-friends"
      >
        Invite
      </button>
      
          
      <button
        class="btn btn-warning cancel-invite"
      >
        Cancel
      </button>
      
      
    </div>
    
  
  </div>
  
</invite-friends>
`;

exports[`Component:  invite component subscribes to friends and updates clear chosen 1`] = `
<invite-friends
  chosen={[Function Array]}
  config={[Function MockConfig]}
  facebookInviteService={[Function MockFacebookInviteService]}
  friendService={[Function MockFriendService]}
  invitable={[Function Array]}
  modalRef={[Function MockModal]}
>
  <div
    class="invite-dialog"
  >
    
      
    <div
      class="modal-header"
    >
      
          
      <h3
        class="modal-title"
      >
        Invite Friends to Play
      </h3>
      
      
    </div>
    
      
    <div
      class="modal-body"
    >
      
          
      <p-multiselect
        class="ng-untouched ng-pristine ng-valid"
        datakey="id"
        maxselectedlabels="30"
        ng-reflect-data-key="id"
        ng-reflect-max-selected-labels="30"
        ng-reflect-model=""
        ng-reflect-option-label="displayName"
        ng-reflect-options="[object Object]"
        optionlabel="displayName"
      >
        
          
        <div
          class="ui-multiselect ui-widget ui-state-default ui-corner-all"
          ng-reflect-ng-class="[object Object]"
        >
          
              
          <div
            class="ui-helper-hidden-accessible"
          >
            
                  
            <input
              readonly="readonly"
              type="text"
            />
            
              
          </div>
          
              
          <div
            class="ui-multiselect-label-container"
            title="Choose"
          >
            
                  
            <label
              class="ui-multiselect-label ui-corner-all"
            >
              Choose
            </label>
            
              
          </div>
          
              
          <div
            class="ui-multiselect-trigger ui-state-default ui-corner-right"
            ng-reflect-ng-class="[object Object]"
          >
            
                  
            <span
              class="ui-clickable fa fa-fw fa-caret-down"
              ng-reflect-klass="ui-clickable"
              ng-reflect-ng-class="fa fa-fw fa-caret-down"
            />
            
              
          </div>
          
              
          <div
            class="ui-multiselect-panel ui-widget ui-widget-content ui-corner-all ui-shadow"
            ng-reflect-ng-class="ui-multiselect-panel ui-widget"
            style="display: none;"
          >
            
                  
            <div
              class="ui-widget-header ui-corner-all ui-multiselect-header ui-helper-clearfix"
              ng-reflect-klass="ui-widget-header ui-corner-all"
              ng-reflect-ng-class="[object Object]"
            >
              
                      
              
              <div
                class="ui-chkbox ui-widget"
              >
                
                          
                <div
                  class="ui-helper-hidden-accessible"
                >
                  
                              
                  <input
                    readonly="readonly"
                    type="checkbox"
                  />
                  
                          
                </div>
                
                          
                <div
                  class="ui-chkbox-box ui-widget ui-corner-all ui-state-default"
                  ng-reflect-klass="ui-chkbox-box ui-widget ui-cor"
                  ng-reflect-ng-class="[object Object]"
                >
                  
                              
                  <span
                    class="ui-chkbox-icon ui-clickable"
                    ng-reflect-klass="ui-chkbox-icon ui-clickable"
                    ng-reflect-ng-class="[object Object]"
                  />
                  
                          
                </div>
                
                      
              </div>
              
                      
              
              <div
                class="ui-multiselect-filter-container"
              >
                
                          
                <input
                  class="ui-inputtext ui-widget ui-state-default ui-corner-all"
                  role="textbox"
                  type="text"
                />
                
                          
                <span
                  class="fa fa-fw fa-search"
                />
                
                      
              </div>
              
                      
              <a
                class="ui-multiselect-close ui-corner-all"
                href="#"
              >
                
                          
                <span
                  class="fa fa-close"
                />
                
                      
              </a>
              
                  
            </div>
            
                  
            <div
              class="ui-multiselect-items-wrapper"
            >
              
                      
              <ul
                class="ui-multiselect-items ui-multiselect-list ui-widget-content ui-widget ui-corner-all ui-helper-reset"
                style="max-height: 200px;"
              >
                
                          
                
                <li
                  class="ui-multiselect-item ui-corner-all"
                  ng-reflect-klass="ui-multiselect-item ui-corner-"
                  ng-reflect-ng-class="[object Object]"
                  style="display: block;"
                >
                  
                              
                  <div
                    class="ui-chkbox ui-widget"
                  >
                    
                                  
                    <div
                      class="ui-helper-hidden-accessible"
                    >
                      
                                      
                      <input
                        aria-label="n1"
                        readonly="readonly"
                        type="checkbox"
                      />
                      
                                  
                    </div>
                    
                                  
                    <div
                      class="ui-chkbox-box ui-widget ui-corner-all ui-state-default"
                      ng-reflect-klass="ui-chkbox-box ui-widget ui-cor"
                      ng-reflect-ng-class="[object Object]"
                    >
                      
                                      
                      <span
                        class="ui-chkbox-icon ui-clickable"
                        ng-reflect-klass="ui-chkbox-icon ui-clickable"
                        ng-reflect-ng-class="[object Object]"
                      />
                      
                                  
                    </div>
                    
                              
                  </div>
                  
                              
                  
                  <label>
                    n1
                  </label>
                  
                              
                  
                  
                          
                </li>
                <li
                  class="ui-multiselect-item ui-corner-all"
                  ng-reflect-klass="ui-multiselect-item ui-corner-"
                  ng-reflect-ng-class="[object Object]"
                  style="display: block;"
                >
                  
                              
                  <div
                    class="ui-chkbox ui-widget"
                  >
                    
                                  
                    <div
                      class="ui-helper-hidden-accessible"
                    >
                      
                                      
                      <input
                        aria-label="n2"
                        readonly="readonly"
                        type="checkbox"
                      />
                      
                                  
                    </div>
                    
                                  
                    <div
                      class="ui-chkbox-box ui-widget ui-corner-all ui-state-default"
                      ng-reflect-klass="ui-chkbox-box ui-widget ui-cor"
                      ng-reflect-ng-class="[object Object]"
                    >
                      
                                      
                      <span
                        class="ui-chkbox-icon ui-clickable"
                        ng-reflect-klass="ui-chkbox-icon ui-clickable"
                        ng-reflect-ng-class="[object Object]"
                      />
                      
                                  
                    </div>
                    
                              
                  </div>
                  
                              
                  
                  <label>
                    n2
                  </label>
                  
                              
                  
                  
                          
                </li>
                
                      
              </ul>
              
                  
            </div>
            
              
          </div>
          
          
        </div>
        
      
      </p-multiselect>
      
      
    </div>
    
      
    <div
      class="modal-footer"
    >
      
          
      <button
        class="btn btn-primary invite-friends"
      >
        Invite
      </button>
      
          
      <button
        class="btn btn-warning cancel-invite"
      >
        Cancel
      </button>
      
      
    </div>
    
  
  </div>
  
</invite-friends>
`;
